packages/react-heartwood-components/src/components/EventDetails/components/EventDetailsItem/EventDetailsItem.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+import List from '../../../List/List'
+import Button from '../../../Button/Button'
+import { CardBuilder } from '../../../Card'
+import Toast from '../../../Toast/Toast'
+import Text from '../../../Text/Text'
+
+export interface IEventDetailsItemProps {
+	/** Unique identifier for the item */
+	id: string
+
+	/** Component key to decide what to render */
+	component: 'list' | 'button' | 'card' | 'toast' | 'text'
+
+	/** Props to pass into the rendered component */
+	// TODO: This should be tied to only the components that can be rendered
```

Comment:
After all are converted to TS right? Might be worth calling that out in the TODO